include/math/matrix.h

```diff
@@ -0,0 +1,83 @@
+#ifndef MATRIX_H
+#define MATRIX_H
+
+#include <math.h>
+
+typedef union {
+    float v[3];
+    struct {
+        float x, y, z;
+    };
+    struct {
+        float r, g, b;
+    };
+} vec3;
```

Comment:
I chose to omit the "`_t`" in the typedef so that it would match the glsl types. But if you prefer it be there we can add it, I don't feel strongly about it.